### PR TITLE
Static checks

### PR DESCRIPTION
**Pull request by pchojnacki**
Static checks look for common mistakes in the code. Running them on every pull request seem like a good idea.

### CODE REVIEW

File: gradle/static_checks.gradle [hunk 1]
```diff
@@ -11,10 +15,10 @@ dependencies {
 }
 
 tasks.withType(FindBugs) {
-    reportLevel = "low"
+//    reportLevel = "low"
```
Comment (nandy-andy) [inline gradle/static_checks.gradle]: Do we need those commented out lines? :japanese_ogre:

Comment (pchojnacki) [inline gradle/static_checks.gradle]: hmm I want to enable them in the future once we have dealt with current crop of errors. So I left the comments as a reminders

Comment (pchojnacki) [inline gradle/static_checks.gradle]: If thats not ok I can remove them

Comment (nandy-andy) [inline gradle/static_checks.gradle]: I think it's OK after given information :smile_cat:  Thanks for clarification :+1:

File: gradle/static_checks.gradle [hunk 2]
```diff
@@ -1,9 +1,13 @@
 apply plugin: 'findbugs'
+apply plugin: 'pmd'
+
 findbugs {
     ignoreFailures = true
 }
 
-apply plugin: 'pmd'
+pmd {
+    ignoreFailures = true
```
Comment (sqreek) [inline gradle/static_checks.gradle]: Please add comments, about why we ignore this failures

File: gradle/static_checks.gradle [hunk 3]
```diff
@@ -1,9 +1,13 @@
 apply plugin: 'findbugs'
+apply plugin: 'pmd'
+
 findbugs {
     ignoreFailures = true
```
Comment (sqreek) [inline gradle/static_checks.gradle]: Please add comments, about why we ignore this failures

Comment (pchojnacki) [inline gradle/static_checks.gradle]: There is 35 errors from findbugs alone on medium settings. Once we fix that and it stops breaking then I can think of removing this

File: gradle/static_checks.gradle (hunk 2 above)
Comment (pchojnacki) [inline gradle/static_checks.gradle]: see comment below

**Comment by sqreek** [comment]
lgtm

**Comment by nandy-andy** [comment]
LGMT :+1:

**Pull request merged**